storybookjs/storybook
# UI: Fix tree highlight performance

## What I did

Various performance tweaks for how items in the sidebar are highlighted. Specifically this avoids rerendering nodes in the tree by using global CSS rather than passing props down. Also we've replaced `throttle` with `requestAnimationFrame` and implemented some other efficiency improvements.

In particular this is noticeable when holding down `ArrowDown` or `ArrowUp` to navigate the tree view. Here's a benchmark in `official-storybook` with `Chromatic` added as a ref (for a total of 2706 nodes), on a 2019 MacBook Pro:

### Before
<img width="947" alt="Screenshot 2020-11-12 at 15 38 18" src="https://user-images.githubusercontent.com/321738/98953610-2e79eb80-24fd-11eb-9adc-d9325b4a58b2.png">

### After
<img width="948" alt="Screenshot 2020-11-12 at 15 32 53" src="https://user-images.githubusercontent.com/321738/98953352-dfcc5180-24fc-11eb-929c-73118e53273a.png">

Where a single keydown event used to take ~205ms and ends up colliding with the next event, it now takes less than 20ms, which is fast enough to not cause any issues.

## How to test

- Is this testable with Jest or Chromatic screenshots? Yes, we've updated the stories, but the snapshots should not change.
- Does this need a new example in the kitchen sink apps? No
- Does this need an update to the documentation? No

If your answer is yes to any of these, please make sure to include it in your PR.
